packages/backend-common/src/config.ts

```diff
@@ -159,12 +194,13 @@ export async function loadBackendConfig(options: {
     configRoot: paths.targetRoot,
     configPaths: configPaths.map(opt => resolvePath(opt)),
     watch: {
-      onChange(newConfigs) {
+      async onChange(newConfigs) {
         options.logger.info(
           `Reloaded config from ${newConfigs.map(c => c.context).join(', ')}`,
         );
 
         config.setConfig(ConfigReader.fromConfigs(newConfigs));
+        await updateRedactionMap(configs, options.logger);
```

Comment:
Let's not build this into the watching logic here imo, but instead have the redaction map logic subscribe to config changes